packages/api-server/api_server/repositories/alerts.py

```diff
@@ -71,13 +72,23 @@ async def acknowledge_alert(self, alert_id: str) -> Optional[ttm.AlertPydantic]:
         # https://github.com/tortoise/tortoise-orm/pull/1131. This is a
         # temporary workaround.
         ack_alert._custom_generated_pk = True  # pylint: disable=W0212
+        unix_millis_acknowledged_time = round(ack_time.timestamp() * 1e3)
         ack_alert.update_from_dict(
             {
                 "acknowledged_by": self.user.username,
-                "unix_millis_acknowledged_time": round(ack_time.timestamp() * 1e3),
+                "unix_millis_acknowledged_time": unix_millis_acknowledged_time,
             }
         )
         await ack_alert.save()
+
+        user = User(username="__rmf_internal__", is_admin=True)
+        task_repo = TaskRepository(user)
+
+        # Save in logs who was the user that acknowledged the task
+        await task_repo.save_log_acknowledged_task_completion(
```

Comment:
Since we are raising exceptions in `save_log_acknowledged_task_completion`, perhaps we should add this within a `try`

Reply:
Done a8bedf464550510432964db1749f71ac745d7aa6 !